docs/Technical-Documentation/clean-and-reparse.md

```diff
@@ -0,0 +1,275 @@
+```
+health status index                   uuid                   pri rep docs.count docs.deleted store.size pri.store.size
+green  open   .kibana_1               VKeA-BPcSQmJJl_AbZr8gQ   1   0          1            0      4.9kb          4.9kb
+yellow open   dev_ssp_m1_submissions  mDIiQxJrRdq0z7W9H_QUYg   1   1          5            0       24kb           24kb
+yellow open   dev_ssp_m2_submissions  OUrgAN1XRKOJgJHwr4xm7w   1   1          6            0     33.6kb         33.6kb
+yellow open   dev_ssp_m3_submissions  60fCBXHGTMK31MyWw4t2gQ   1   1          8            0     32.4kb         32.4kb
+yellow open   dev_tanf_t1_submissions 19f_lawWQKSeuwejo2Qgvw   1   1        817            0    288.2kb        288.2kb
+yellow open   dev_tanf_t2_submissions dPj2BdNtSJyAxCqnMaV2aw   1   1        884            0    414.4kb        414.4kb
+yellow open   dev_tanf_t3_submissions e7bEl0AURPmcZ5kiFwclcA   1   1       1380            0    355.2kb        355.2kb
+```
+All tests are considered to have been run INDEPENDENTLY. For each test, your Elastic and DAC state should match the initial conditions above. The commands in the section below should be run in between each test if you want to match the expected output.
+
+#### Some Useful Commands to Reset Elastic State
+The commands should ALWAYS be executed in the order they appear below.
+1. curl -X DELETE 'http://localhost:9200/dev*'
+2. python manage.py search_index --rebuild
+
+#### Clean and Re-parse All with New Indices and Keeping Old Indices
+1. Execute `python manage.py clean_and_reparse -a -n`
+    - If this is the first time you're executing a command with new indices, because we have to create an alias in Elastic with the same name as the
+    original index i.e. (`dev_tanf_t1_submissions`), the old indices no matter whether you specified `-d` or not will be deleted. From thereafter,
+    the command will always respect the `-d` switch.
+2. Expected Elastic results.
+    - If this is the first time you have ran the command the [indices](http://localhost:9200/_cat/indices/?pretty&v&s=index) url should reflect 21 indices prefixed with `dev` and they should contain the same number of documents as the original indices did. The new indices will also have a datetime suffix indicating when the re-parse occurred.
+    - If this is the second time running this command the [indices](http://localhost:9200/_cat/indices/?pretty&v&s=index) url should reflect 42 indices prefixed with `dev` and they should each contain the same number of documents as the original indices did. The latest indices will have a new datetime suffix delineating them from the other indices.
+3. Expected DAC results.
+    - The DAC record counts should be exactly the same no matter how many times the command is run.
+    - The primary key for all reparsed datafiles should no longer be the same.
+    - `ParserError` and `DataFileSummary` objects should be consistent with the file.
+
+- First run:
```

Comment:
Let's cut the long outputs from these runs. The initial local output starting L40 will give the reader a sense of what to look at/for.